src/containers/Navigation/index.tsx

```diff
@@ -21,21 +27,32 @@ const Navigation = ({ sections, currentSlug, label }: Props): JSX.Element => {
     }
   };
 
-  const autoScroll = async (height: number): Promise<void> => {
-    if ((isOpen || !isMobileScreen()) && !hasScrolled && navElement.current) {
+  if (typeof window !== 'undefined') {
+    const nav = navElement.current;
+
+    if (nav) {
+      const CurrentLinkElem = document.getElementById(`link-${currentSlug}`);
       const { newScrollPos, scrollWindow, scrollTime } = calcNavScrollParams(
-        height,
-        navElement.current
+        CurrentLinkElem?.getBoundingClientRect().top || 0,
+        nav
       );
-
-      try {
-        await scrollTo(newScrollPos, scrollWindow, scrollTime);
-        setHasScrolled(true);
-      } catch (e) {
-        // TODO: follow up with appropriate error logging if any
-        setHasScrolled(false);
+      if (!previousSlug) {
+        nav.scrollTop = newScrollPos;
+      } else {
+        const PrevLinkElem = document.getElementById(`link-${previousSlug}`);
+        const prevElemScroll = calcNavScrollParams(
+          PrevLinkElem?.getBoundingClientRect().top || 0,
+          nav
+        );
+        nav.scrollTop = prevElemScroll.newScrollPos;
+        scrollTo(newScrollPos, scrollWindow, scrollTime);
       }
     }
+  }
+
+  const reRender = async (): Promise<void> => {
```

Comment:
Oh, and also, maybe we could think about more descriptive name because _reRender_ in React is somewhat vague, a lot of actions trigger rerender.

Reply:
The rerender function was something I had to stitch together for this to work. I'm thinking to remove it completely, keeping the logic only to the Navigation Container. I think that'll make it more universal